src/Tizen.NUI/src/public/Capture.cs

```diff
@@ -0,0 +1,318 @@
+                // dali native definition :
+                // enum class FinishState
+                // {
+                //   SUCCEEDED, ///< Succeeded in saving the result after capture
+                //   FAILED     ///< Failed to capture by time out or to save the result
+                // };
+                arg.Success = (state == 0) ? true : false;
+                finishedEventHandler?.Invoke(this, arg);
+            }
+        }
+
+        private CaptureSignal finishedSignal;
+        private delegate void finishedCallbackType(IntPtr data, int state);
+        private finishedCallbackType finishedCallback;
+        private EventHandler<CaptureFinishStateEventArgs> finishedEventHandler;
+
+        /// <summary>
+        /// For subscribing Finished event sent by this class.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<CaptureFinishStateEventArgs> Finished
+        {
+            add
+            {
+                if (finishedEventHandler == null && disposed == false)
+                {
+                    finishedSignal = new CaptureSignal(Interop.Capture.Get(swigCPtr), false);
+                    finishedCallback = onFinished;
+                    finishedSignal.Connect(finishedCallback);
+                }
+                finishedEventHandler += value;
+            }
+            remove
+            {
+                finishedEventHandler -= value;
+
+                if (finishedEventHandler == null && finishedSignal?.Empty() == false)
+                {
+                    finishedCallback = onFinished;
+                    finishedSignal.Disconnect(finishedCallback);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get NativeImageSource that is saved captured image.
+        /// </summary>
+        /// <returns>NativeImageSource that is saved captured image.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public NativeImageSource GetNativeImageSource()
+        {
+            Tizen.Log.Debug("NUI", $"GetNativeImageSource()");
+            return new NativeImageSource(Interop.Capture.GetNativeImageSourcePtr(swigCPtr), true);
+        }
+
+        /// <summary>
+        /// Generate captured image's Url
+        /// </summary>
+        /// <returns>The Url string representing this captured image source</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string GenerateUrl()
+        {
+            string url = "";
+            url = Interop.Capture.GenerageUrl(swigCPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return url;
+        }
+
+    }
+
+    /// <summary>
+    /// CaptureFinishStateEventArgs
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class CaptureFinishStateEventArgs : EventArgs
```

Comment:
CaptureFinishedEventArgs is commonly used style. please do double check.

Reply:
yes. you are right. I changed it as CaptureFinishedEventArgs.